nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,372 @@
+            else:
+                results_submitted_date = None
+
+            #last_updated_date 
+            lud = (soup.last_update_submitted).text
+            last_updated_date = datetime.strptime(lud, '%B %d, %Y').date()
+
+            #certificate_date
+            if hc is not None:
+                certificate_date = datetime.strptime(hc.text, '%B %d, %Y').date()
+            else:
+                certificate_date = None
+
+            #enrollment
+            enrollment = (soup.enrollment).text
+
+            #sponsor
+            sponsor = (soup.sponsors.lead_sponsor.agency).text
+
+            #sponsor_type
+            sponsor_type = (soup.sponsors.lead_sponsor.agency_class).text
+
+            #collaborators
+            try:
+                collaborators = json.dumps(parsed_json['clinical_study']['sponsors']['collaborator'])
+            except KeyError:
+                collaborators = None
+
+            #exported
+            if soup.oversight_info.is_us_export is not None:
+                exported = (soup.oversight_info.is_us_export).text
+            else:
+                exported = None
+
+            #url
+            url = (soup.url).text
+
+            #official_title
+            if soup.official_title is None:
+                official_title = None
+            else:
+                official_title = (soup.official_title).text
+
+            #brief_title
+            if soup.brief_title is None:
+                brief_title = None
+            else:
+                brief_title = (soup.brief_title).txt
+
+            #title
+            if official_title is not None:
+                title = official_title
+            elif official_title is None and brief_title is not None:
+                title = brief_title
+            else:
+                title = None
+
+            #discrep_date_status - add this to "if" statement after testing: 'and completion_date is not null')
+            if (primary_completion_date < date.today() or primary_completion_date is None) and\
+            completion_date < date.today() and\
+            study_status is ("Unknown status" or "Active, not recruiting" or "Not yet recruiting" or\
+            "Enrolling by invitation" or "Suspended" or "Recruiting"):
+                discrep_date_status = 1
+            else:
+                discrep_date_status = 0
+
+            #late_cert
+            if certificate_date is not None and certificate_date > (certificate_date + relativedelta(years=1)):
+                late_cert = 1
+            else:
+                late_cert = 0
+
+            #defaulted_date
+            if (used_primary_completion_date == 1 and defaulted_pcd_flag == 1) or\
+            used_primary_completion_date == 0 and defaulted_cd_flag == 1:
+                defaulted_date = 1
+            else:
+                defaulted_date = 0
+
+            #condition
+            try:
```

Comment:
Make a little function to avoid repeating identical try/excepts here and below: `json_or_none()` or similar.